src/suites/cts/gpu_shader_test.ts

```diff
@@ -0,0 +1,31 @@
+import { GPUTest } from './gpu_test.js';
+
+// tslint:disable-next-line: no-any
+let glslangInstance: any = undefined;
+
+export class GPUShaderTest extends GPUTest {
+  async init(): Promise<void> {
+    await super.init();
+    if (!glslangInstance) {
+      const glslangPath = '../../glslang.js';
+      const glslangModule = (await import(glslangPath)).default;
+      await new Promise(resolve => {
+        // tslint:disable-next-line: no-any
+        glslangModule().then((module: any) => {
+          glslangInstance = module;
+          resolve();
+        });
+      });
+    }
+  }
+
+  makeShaderModule(type: 'f' | 'v' | 'c', source: string): GPUShaderModule {
+    return this.device.createShaderModule({ code: this.compile(type, source) });
+  }
+
+  private compile(type: 'f' | 'v' | 'c', source: string): Uint32Array {
+    const t = type === 'f' ? 'fragment' : type === 'v' ? 'vertex' : 'compute';
```

Comment:
nit: If the compiler takes the full string names, let's just use the full name as the `type` argument for better readability.

Reply:
Done.